### PR TITLE
Refurbish installation instructions

### LINKED ISSUE #677: Simplify installation instructions (datalad-installer)
The installer is now on PyPi and we use it for CIs on all platforms. There is definitely the potential for a more compact, more long-term stable installation documentation. We could switch from lots of custom lines to the description of installation scenarios supported by the installer. This includes windows.

### PR DESCRIPTION
**Pull request by mih**
Incorporating the latest wisdom and helpers.

The principle idea is to go from specific to generic, in order to let
readers with less expertise find working solutions more quickly, while
not hiding alternatives with better system-integration potential for
more experienced users.

Fixes #677 

@yarikoptic @jwodder If you have advice on what to do differently, please let me know.

### CODE REVIEW

File: docs/intro/installation.rst [hunk 1]
```diff
@@ -100,113 +295,110 @@ available version.
 Linux: CentOS, Redhat, Fedora, or similar systems
 """""""""""""""""""""""""""""""""""""""""""""""""
 
-For CentOS, Redhat, Fedora, or similar distributions, there is an rpm
-git-annex-standalone available
-`here <https://git-annex.branchable.com/install/rpm_standalone/>`_.
-Subsequently, DataLad can be installed via ``pip``.
+For CentOS, Redhat, Fedora, or similar distributions, there is an `RPM package
+for git-annex <https://git-annex.branchable.com/install/rpm_standalone/>`_.  A
+suitable version of Python and :term:`Git`, should come with the operating
```
Comment (adswa) [inline docs/intro/installation.rst:300]:
```suggestion
for git-annex <https://git-annex.branchable.com/install/rpm_standalone/>`_.  A
suitable version of Python and :term:`Git` should come with the operating
```

stray comma

**Review by adswa — APPROVED** [review]
Thank you! This structures this section much better and with less duplication, and it adds quite a number of useful information, especially about the datalad installer! I have added comments with a few minor language/typo nitpicks.

File: docs/intro/installation.rst [hunk 2]
```diff
@@ -3,92 +3,287 @@
...
+   your system. Python 2 is an outdated, in technical terms "deprecated",
+   version of Python.  Although it still exist as the default Python version on
+   many systems, it is no longer maintained since 2020, and thus, most software
+   has dropped support for Python 2.  If you only run Python 2 on your system,
+   most Python software, including DataLad, will be incompatible, and hence
+   unusable, resulting in errors during installation and execution.
+
+   The easiest way to verify that this is the case is to open a terminal and
+   type :command:`python` to start a Python session::
+
```
[the remark this answers is not in the document]
Comment (mih) [reply docs/intro/installation.rst:40]: Good point! Maybe moving all of Python2 to the end is another approach -- this info will become less and less relevant over time.

[the remark this answers is not in the document]
Comment (adswa) [reply docs/intro/installation.rst:40]: True. This is a good idea :+1:

[the remark this answers is not in the document]
Comment (mih) [reply docs/intro/installation.rst]: I pushed an alternative.

**Comment by mih** [comment]
@yarikoptic @jwodder I will merge this now, to be able to continue in related work, but please leave your comments, if you see potential for better. Thx!

**Comment by mih** [comment]
Thx for the review @adswa !

**Pull request merged**